app/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM node:11-stretch-slim
+
+COPY . /usr/local/src
+WORKDIR /usr/local/src
+
+RUN set -ex \
+	&& buildDeps=" \
+        python \
+        make \
+        build-essential \
+	" \
+	&& apt-get update && apt-get install -y $buildDeps --no-install-recommends \
+	&& yarn install \
+	&& apt-get purge -y --auto-remove $buildDeps \
+	&& rm  -Rf /var/lib/apt/lists/*
```

Comment:
This block is annoyingly verbose, but the gist is that we want to wrap the installation step in `apt` commands for installing and then purging required build dependencies because we don't actually want the build dependencies to show up in the final image. (If we installed and then deinstalled in separate layers, it would still increase the size of the image in spite of not including the dependencies in the final image layer.) This is a pretty common Dockerfile pattern but I've never read anything about it; if it'd be helpful I can try to dig up some resources.